python/pynessie/commands/merge.py

```diff
@@ -70,5 +73,23 @@ def merge(ctx: ContextObject, ref: str, force: bool, expected_hash: str, hash_on
             """Either condition or force must be set. Condition should be set to a valid hash for concurrency
             control or force to ignore current state of Nessie Store."""
         )
-    ctx.nessie.merge(from_ref, ref, hash_on_ref, expected_hash)
-    click.echo()
+    merge_response = ctx.nessie.merge(from_ref, ref, hash_on_ref, expected_hash)
+    if merge_response is None:
+        if ctx.json:
+            hint = {"hint": "Merge succeeded but legacy server did not respond with additional details."}
+            click.echo(json.dumps(hint))
+        else:
+            click.echo("Merge succeeded but legacy server did not respond with additional details.")
+    else:
+        if ctx.json:
+            click.echo(MergeResponseSchema().dumps(merge_response))
+        else:
+            commits = merge_response.source_commits
+            click.echo(
+                "The following {N} commits were merged onto {target_branch}:".format(
```

Comment:
we should mention the commit hash for the merge? i.e. the hash on the target branch after this operation was done
(think about many operations happening concurrently and you need to decipher the order of things from CLI log output afterwards)

also since you mentioned that commits could be squashed, maybe this info is also available in the response somehow and we should surface it?